reactor-core/src/test/java/reactor/test/publisher/BaseOperatorTest.java

```diff
@@ -595,20 +595,14 @@ final void inputFusedAsyncOutputFusedAsyncCancel(OperatorScenario<I, PI, O, PO>
 			Sinks.Many<I> up = Sinks.unsafe().many().unicast().onBackpressureBuffer();
 			testUnicastSource(scenario, up);
 			StepVerifier.create(scenario.body()
-			                            .apply(withFluxSource(up.asFlux())), 0)
+			                            .apply(withFluxSource(up.asFlux())), scenario.producing)
 			            .consumeSubscriptionWith(s -> {
 				            if (s instanceof Fuseable.QueueSubscription) {
 					            @SuppressWarnings("unchecked")
 					            Fuseable.QueueSubscription<O> qs = ((Fuseable.QueueSubscription<O>) s);
 					            qs.requestFusion(ASYNC);
-					            //UnicastProcessor#actual
-					            if (up.scan(Attr.ACTUAL) != qs || scenario.prefetch() == -1) {
-						            qs.size(); //touch undeterministic
-					            }
-					            else {
-						            assertThat(qs.size()).isEqualTo(up.scan(Attr.BUFFERED)); //UnicastProcessor#size
-					            }
 					            try {
+									// Assuming scenario.producing >= 3
```

Comment:
This seems to have been a mistake as on line 598 the demand was 0. So it was actually illegal to poll, which causes onNext signals to be triggered. In cases such as filtering out all the values, this would trigger upstream demand to continuously increase forcing termination. Therefore, the initial demand is set to scenario.producing and this code assumes it's not less than 3 to meet the three polls that happen here.

Reply:
Actually, as discussed with @OlegDokuka `poll`-ing expresses demand on its own, so can be an optimization technique in these fused scenarios and not really a misuse. The issue observed was with not protecting against simultaneous `onNext` that can happen as a result of calling `poll` during `onSubscribe`. Now with proper protection in place, the problem does not occur.